### ISSUE TITLE
[question] Nodes disappearing / Cannot re-include 

### ISSUE THREAD

**Issue by linux-guru-com** [zwave-js/zwave-js-ui#2677]
I am on the (as of now) latest docker image zwavejs/zwavejs2mqtt:latest which provides zwave-js-ui: 8.0.0 zwave-js: 10.2.0 (in cooperation with Home Assistant 202209.6).
Out of a sudden I am not able to re-include failed devices after I did a factory default reset.
Actions->Manage Nodes->Inclusion->->No encryption results in a screen showing "Inclusion started: XXs remaining" where XX is anything between 15 and 0 (different on each attempt). The screen stays forever.
The device does never appear again on the overview and, even worse, after rebooting the whole host (RPI3B) it is very likely that another device disappears from the device list. Sometimes it comes back after another reboot.
My hardware:
RPI3B, AEON Labs ZW090, Danfoss Devolo Thermostat (09356) MT02650.
For the thermostat the vendor documentation states, that for a factory reset, the middle button needs to be pressed for at least 5 seconds after re-inserting the batteries. This then leads into a fixing/unfixing procedure of the thermostat head and a flashing display. Pressing the middle button again, it starts again to flash high frequently at first and with a lower frequence 2-3 seconds later until the display light turns off.
Vendor documentation states: "Inclusion
Click on the middle button will confirm inclusion or exclusion and wakeup the device for wireless communication. A long push for 3 seconds on the middle buttons enters and leaves the management mode indicated by a "M"."
That's exactly the procedure I followed.
The issue is re-producable with other devices of the same type/model.

If I want to exclude or remove failed nodes, I am running into errors as well:
"Failed Nodes/Remove all" results in "Error while calling api removeFailedNode: The node removal process could not be started because the node responded to a ping. (ZW0360)"
Actions->Manage Nodes->Exclusions --> Pressing the button on the device results in "Exclusion stopped, no changes detected"

Config and logs are attached.

Is anybody able to help or support here so I can get my thermostates back working again? 
It would be pleasant to not sit in a cold house just because of such a "minor" issue... ;-)

[zwave-js-ui-store.zip](https://github.com/zwave-js/zwave-js-ui/files/9638671/zwave-js-ui-store.zip)
[nodes_dump.json.txt](https://github.com/zwave-js/zwave-js-ui/files/9638677/nodes_dump.json.txt)
[nodes.json.txt](https://github.com/zwave-js/zwave-js-ui/files/9638678/nodes.json.txt)

**Comment by robertsLando** [comment]
Seems that for some reason I keep tring to reconnect to the driver even if there is a running connection. Everything starts here:

```
2022-09-24 13:10:47.012 INFO Z-WAVE: Network scan complete. Found: 11 nodes
2022-09-24 13:10:48.308 INFO Z-WAVE: Next update scheduled for: Sun Sep 25 2022 00:00:00 GMT+0200 (Central European Summer Time)
2022-09-24 13:10:54.479 INFO Z-WAVE: Controller status: Driver: Failed to open the serial port: Error Resource temporarily unavailable Cannot lock port (ZW0100)
2022-09-24 13:10:54.483 WARN Z-WAVE: Retry connection in 3 seconds...
```

When `Failed to open the serial port: Error Resource temporarily unavailable Cannot lock port (ZW0100)` happens I start retring to call connect indeterminately:

```
2022-09-24 13:21:35.347 WARN Z-WAVE: Retry connection in 3 seconds...
2022-09-24 13:21:38.371 INFO Z-WAVE: Connecting to /dev/zwave
2022-09-24 13:21:47.428 WARN Z-WAVE: Retry connection in 3 seconds...
2022-09-24 13:21:50.452 INFO Z-WAVE: Connecting to /dev/zwave
2022-09-24 13:21:59.506 WARN Z-WAVE: Retry connection in 3 seconds...
2022-09-24 13:22:02.533 INFO Z-WAVE: Connecting to /dev/zwave
2022-09-24 13:22:11.584 WARN Z-WAVE: Retry connection in 3 seconds...
2022-09-24 13:22:14.610 INFO Z-WAVE: Connecting to /dev/zwave
2022-09-24 13:22:23.666 WARN Z-WAVE: Retry connection in 3 seconds...
2022-09-24 13:22:26.703 INFO Z-WAVE: Connecting to /dev/zwave
2022-09-24 13:22:35.761 WARN Z-WAVE: Retry connection in 3 seconds...
```


@AlCalzone Any clue?

**Comment by AlCalzone** [comment]
Yeah something seems iffy with the connection to the stick. Maybe the two addons fighting over control. Only one can be running at the same time.

**Comment by robertsLando** [comment]
@AlCalzone what do you mean? Maybe he has configured both hass integration and zwavejs2mqtt to connect to the stick?

**Comment by AlCalzone** [comment]
yes

**Comment by robertsLando** [comment]
@linux-guru-com ? You should only configure zwave-js-ui to connect to the stick

**Comment by linux-guru-com** [comment]
Thanks for looking into this issue.

I didn't configure it elsewhere.
Setup is:
RPI3 -> Zwave Stick -> Docker Image (mapping /dev/ttyACM0:/dev/zwave) -> 
zwave-js/zwave-js-ui -> Websocket -> HASS (PyPi directly on RPI via 
venv) -> ZwaveJS-Integration

Maybe you did see the logs from the time where I tried to solve the 
issue on my own and where I switched from /dev/ttyACM0 to 
/dev/serial/device-by-id/xxx?

Today's log doesn't show this device locking issue anymore...

2022-09-25T22:02:02.907Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-25T22:35:24.998Z CNTRLR   [Node 006] Timed out while waiting for 
a response from the node (ZW0201)
2022-09-26T00:09:03.443Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T00:13:56.539Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T02:51:21.461Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T03:40:34.996Z CNTRLR   [Node 006] Timed out while waiting for 
a response from the node (ZW0201)
2022-09-26T03:40:35.024Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T03:55:20.017Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T04:18:14.129Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T04:20:11.085Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T04:25:06.674Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T04:39:38.316Z CNTRLR   [Node 006] Timed out while waiting for 
a response from the node (ZW0201)
2022-09-26T05:04:16.058Z CNTRLR   [Node 006] Timed out while waiting for 
a response from the node (ZW0201)
2022-09-26T05:28:52.431Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T06:13:16.111Z CNTRLR   [Node 006] did not respond after 1/3 
attempts. Scheduling next try in 500 ms.
2022-09-26T06:13:16.749Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T06:14:33.614Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.
2022-09-26T06:18:11.455Z CNTRLR   [Node 006] The node did not respond 
after 1 attempts. It is probably asleep, m
                                   oving its messages to the wakeup queue.
2022-09-26T06:18:11.484Z CNTRLR   [Node 006] error during API call: 
Failed to send the command after 1 attempts
                                   (Status NoAck) (ZW0204)
2022-09-26T06:18:21.497Z CNTRLR   [Node 006] error during API call: The 
message has expired (ZW0213)
2022-09-26T06:37:49.529Z CNTRLR   Failed to execute controller command 
after 1/3 attempts. Scheduling next try i
                                   n 100 ms.


Do you have any other idea? ;-)
Cheers


Am 26.09.2022 um 09:28 schrieb Daniel Lando:
> @linux-guru-com <https://github.com/linux-guru-com> ? You should only 
> configure zwave-js-ui to connect to the stick

**Comment by AlCalzone** [comment]
> /dev/ttyACM0:/dev/zwave

or 

> /dev/serial/device-by-id/xxx

which are you using?

---

The logs are not on loglevel debug, so they aren't helpful. And please **attach** them by dragging and dropping a file into the text field.

**Comment by linux-guru-com** [comment]
/dev/ttyACM0:/dev/zwave is the one I am using.

/dev/serial/device-by-id/xxx was just a try which lead into the logged 
locking issue.
Actually it was /dev/serial/by-id/xxx (without the word 'device')...

 From the system:
# ll /dev/ttyACM0
crw-rw---- 1 root dialout 166, 0 Sep 24 13:53 /dev/ttyACM0

# ll /dev/serial/by-id/usb-0658_0200-if00
lrwxrwxrwx 1 root root 13 Sep 24 13:53 
/dev/serial/by-id/usb-0658_0200-if00 -> ../../ttyACM0

Am 26.09.2022 um 10:22 schrieb AlCalzone:
>     /dev/ttyACM0:/dev/zwave
> 
> or
> 
>     /dev/serial/device-by-id/xxx
> 
> which are you using?
>

**Comment by linux-guru-com** [comment]
Maybe another helpful information:
Yesterday evening it came to my mind, that I did not factory-reset one of the failed thermostats and I did simply re-include it (without resetting) and it worked without any issues.
This somehow makes me believing that we either have an issue with integration from scratch (first-time integration of an unknown device) or even have some information left somewhere in the config (how? where?) which makes the re-inclusion procedure to fail.
What do you think about these ideas?

**Comment by robertsLando** [comment]
> Yesterday evening it came to my mind, that I did not factory-reset one of the failed thermostats and I did simply re-include it (without resetting) and it worked without any issues.

This is generally a bad idea, how did you managed to do it without resetting the device? If it was already included it shouldn't allow to be included again without firstly excluding it or factory resetting the device

**Comment by linux-guru-com** [comment]
I just pressed the middle button on the thermostat for waking it up and 
started the include process via the GUI. Did work like a charm... :-o
The device got the same ID than again as it used to have and appeared in 
HASS again as it would have never been away...

**Comment by AlCalzone** [comment]
Ok...
1. Prefer the by-id path over acm0. It's just a symlink but it is stable, even if the target path accidentally changes

2. Including an already included device can work, we can ignore this for now

3. Logs 🙃

**Comment by linux-guru-com** [comment]
You mean I should generate some (new) logs?
If so, should I also retry to integrate the failed devices once more 
while having the loglevel on "silly"?

**Comment by AlCalzone** [comment]
Yes, like I wrote in my other comment https://github.com/zwave-js/zwave-js-ui/issues/2677#issuecomment-1257671997
Make it loglevel "debug" please.

**Comment by linux-guru-com** [comment]
After changing the loglevel to Debug and re-performing the same procedure(s) on the thermostats as I did in the last 2 days, Inclusion worked properly for all devices.
I have no idea, if this is a coincidence to changing the loglevel or some "cosmic x-ray caused by sun erruption" (aka BOFH excuse) has been the reason, but it is finally working properly.
Thanks for your great support! ;-)

**Comment by robertsLando** [comment]
Feel free to reopen in case of the error comes back :)
